packages/mask/src/utils/network-schema.ts

```diff
@@ -36,14 +37,32 @@ export async function fetchChains(): Promise<ChainConfig[]> {
 
 type NameValidator = (name: string) => boolean | Promise<boolean>
 
+const getRpcChainId = memoize(async (rpc: string) => {
+    const res = await fetch(rpc, {
+        method: 'post',
+        headers: {
+            'Content-Type': 'application/json',
+            Origin: location.origin,
+        },
+        body: JSON.stringify(
+            createJsonRpcPayload(0, {
+                method: EthereumMethodType.ETH_CHAIN_ID,
+                params: [],
+            }),
+        ),
+    })
+    const json = (await res.json()) as { result: string }
+    return Number.parseInt(json.result, 16)
+})
+
 /**
  * schema with basic validation
  * duplicated name validator is injected as dependency, both frontend and
  * background can provide their own validator
  */
 export function createBaseSchema(t: I18NFunction, duplicateNameValidator: NameValidator) {
```

Comment:
Please refactor to use `packages/web3-providers/src/Web3/EVM/schemas/ChainDescriptor.ts`.